docs: add getting started + updated contribution guide (#58)
surrealdb/surrealdb.js: README.md

The official SurrealDB library for JavaScript.

[![npm version](https://badge.fury.io/js/surrealdb.js.svg)](https://badge.fury.io/js/surrealdb.js)
[![deno module](https://shield.deno.dev/x/surrealdb)](https://deno.land/x/surrealdb)

[![](https://img.shields.io/badge/status-beta-ff00bb.svg?style=flat-square)](https://github.com/surrealdb/surrealdb.js)
[![](https://img.shields.io/badge/docs-view-44cc11.svg?style=flat-square)](https://surrealdb.com/docs/integration/libraries/javascript)
[![](https://img.shields.io/badge/license-Apache_License_2.0-00bfff.svg?style=flat-square)](https://github.com/surrealdb/surrealdb.js)

## Quickstart-Guide

### Install

#### Deno

Import it with

```ts
import Surreal from "https://deno.land/x/surrealdb/mod.ts";
```

> Note you should set a version in the url! For example
> `https://deno.land/x/surrealdb@0.5.0/mod.ts`

> #### NodeJS, Browser with bundler

Reviewer: ```suggestion
#### NodeJS, or browser with a bundler
```

> For NodeJS or Browser with bundler like rollup, vite, webpack, ... you can just

Reviewer: ```suggestion
For NodeJS or a browser with bundler (for example: rollup, vite, or webpack) you can just:
```

install the npm package:

```sh
npm i surrealdb.js
# or with yarn
yarn add surrealdb.js
# or with pnpm
pnpm i surrealdb.js
```

> just import it with

Reviewer: ```suggestion
then, just import it with:
```

```ts
import Surreal from "surrealdb.js";
```

> or when you use commonjs

Reviewer: ```suggestion
or when you use CommonJS
```

```ts

> const Surreal = require("surrealdb.js");

Reviewer: Small code fix:
```suggestion
const Surreal = require("surrealdb.js").default;
```

Reviewer 2: This helped through my struggle to make this thing work. Thanks

> const Surreal = require("surrealdb.js").default;

Reviewer 3: ```suggestion
const { default: Surreal } = require("surrealdb.js");
```

		title: "Founder & CEO",
		name: {
			first: "Tobie",
			last: "Morgan Hitchcock",
		},
		marketing: true,
		identifier: Math.random().toString(36).substr(2, 10),
	});

	// Update a person record with a specific id
	let updated = await db.change("person:jaime", {
		marketing: true,
	});

	// Select all people records
	let people = await db.select("person");

	// Perform a custom advanced query
	let groups = await db.query(
		"SELECT marketing, count() FROM type::table($tb) GROUP BY marketing",
		{
			tb: "person",
		},
	);
} catch (e) {
	console.error("ERROR", e);
}
```

## More informations

The docs of this libary are located at
[https://surrealdb.com/docs/integration/libraries/javascript](https://surrealdb.com/docs/integration/libraries/javascript)

## Contribution notes

### Local setup

This is a [Deno](https://deno.land) project, not NodeJS. For example, this means
import paths include the `.ts` file extension. However, to also support other
JavaScript environments, a build has been added to create a npm package that
works for NodeJS, Bun, browsers with bundlers.

#### Supported environments

- [Deno](https://deno.land)
- [NodeJS](https://nodejs.org)
- [Bun](https://bun.sh)

> - web-browsers

Reviewer: ```suggestion
- Web Browsers
```
